app/src/main/java/ch/sdp/vibester/GenreTemporary.kt

```diff
@@ -31,20 +31,16 @@ class GenreTemporary : AppCompatActivity() {
 
     /**
      * Fetch data from Lastfm and show song list in a ListView
-     * @param method: BY_TAG or BY_CHART (top tracks without tag)
-     * @param tag: tag name if method BY_TAG is chosen
      */
     fun performQuery(uri:LastfmUri){
 
-        val service = ServiceBuilder.buildService(baseurl, LastfmApiInterface::class.java)
+        val service = ServiceBuilder.buildService(baseUrl, LastfmApiInterface::class.java)
         val call = service.getSongList(uri.convertToHashmap())
-        call.enqueue(object: Callback<Object> {
-            override fun onFailure(call: Call<Object>, t: Throwable?) {}
+        call.enqueue(object: Callback<Any> {
+            override fun onFailure(call: Call<Any>, t: Throwable?) {}
 
-            override fun onResponse(call: Call<Object>, response: Response<Object>) {
-                if (response != null) {
-                    switchToGame(Gson().toJson(response.body()), uri.method)
-                }
+            override fun onResponse(call: Call<Any>, response: Response<Any>) {
+                switchToGame(Gson().toJson(response.body()), uri.method)
```

Comment:
We have different handling of response from retrofit. Here now we dont check if the response is null or not, in LyricBelondGameActivity we do check for null.  I am not sure if the null check is needed in other places as well. I believe response has an HTTP response code in success/failure, so it cant be null at all?

Reply:
so instead of null, we need to check whether the response was successful or not, and do appropriate actions? If this is the case, I will handle it on my part next PR

Reply:
It was the case before, but somehow my IDE said it was unnecessary and suggested I simplify it to the current version. It seems that sometimes the check is important sometimes not that much, not sure if it’s because the type is Any here.